### PR TITLE
[v3] avoid focus-visible style being cut off by overflow hidden

### PR DESCRIPTION
**Pull request by 87xie**
fixes https://github.com/shuding/nextra/issues/1432

## Description

I noticed some adjustments to the focus-visible style in [Auth.js](https://authjs.dev/) related to Nextra, such as:

- [tab button](https://github.com/nextauthjs/next-auth/blob/538333d86e93e0e4019d117d07741e02f9581c2b/docs/pages/global.css#L177-L180)
- [sidebar](https://github.com/nextauthjs/next-auth/blob/538333d86e93e0e4019d117d07741e02f9581c2b/docs/pages/global.css#L207-L210)

I think these adjustments could be applied more comprehensively within Nextra. For instance, Auth.js only targets the focus-visible style of the first-level sidebar items and the focus-visible style is not working well in the TOC either.

image

image

image

## Screenshots

**Sidebar**

image

image

**Tabs**

image

**TOC**

image

image

**Direction: rtl**

image

image

### CODE REVIEW

File: examples/swr-site/theme.config.tsx [hunk 1]
```diff
@@ -191,7 +191,7 @@ const config: DocsThemeConfig = {
   },
   toc: {
     extraContent: (
-      <img alt="placeholder cat" src="https://placekitten.com/g/300/200" />
+      <img alt="placeholder cat" src="https://placecats.com/300/200" />
```
Comment (87xie) [inline examples/swr-site/theme.config.tsx:194]: In `example/swr-site`, placekitten seems to be down, so I replaced it with placecats.

File: packages/nextra-theme-docs/src/components/sidebar.tsx [hunk 2]
```diff
@@ -400,22 +400,18 @@ export function Sidebar({
           </div>
         )}
         <FocusedItemContext.Provider value={focused}>
-          <OnFocusItemContext.Provider
-            value={item => {
-              setFocused(item)
-            }}
-          >
+          <OnFocusItemContext.Provider value={setFocused}>
```
Comment (87xie) [inline packages/nextra-theme-docs/src/components/sidebar.tsx:403]: It seems that the extra inline function is not needed.

File: packages/nextra-theme-docs/src/components/sidebar.tsx [hunk 3]
```diff
@@ -183,7 +183,7 @@ function FolderImpl({ item, anchors }: FolderProps): ReactElement {
           )}
         />
       </ComponentToUse>
-      <Collapse className="ltr:_pr-0 rtl:_pl-0 _pt-1" isOpen={open}>
+      <Collapse className="_p-1" isOpen={open}>
```
Comment (87xie) [inline packages/nextra-theme-docs/src/components/sidebar.tsx:186]: To resolve a situation like this 

image

**Comment by dimaMachina** [comment]
closed by https://github.com/shuding/nextra/pull/3160, thanks for the initial attempt 🙌

**Pull request closed without merging**